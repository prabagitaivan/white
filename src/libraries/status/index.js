export const isLightMode = () => {
  const hour = new Date().getHours()

  if (hour >= 6 && hour <= 17) return true
  else return false
}

export const isMacPlatform = () => {
  // use deprecated version as the new one still have minim browser support
  // https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/platform
  const platform = window.navigator.platform || ''

  return platform.indexOf('Mac') === 0
}
