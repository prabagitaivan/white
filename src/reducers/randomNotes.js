import { createActions, handleActions } from 'redux-actions'

const defaultState = {
  data: [],
  requesting: true,
  error: null
}

export const {
  randomNotes: { request, success, failure }
} = createActions({
  RANDOM_NOTES: {
    REQUEST: action => action,
    SUCCESS: action => action,
    FAILURE: action => action
  }
})

export default handleActions(
  {
    [request]: state => ({ ...state, requesting: true, error: null }),
    [success]: (state, { payload: data = [] }) => ({
      ...state,
      data,
      requesting: false,
      error: null
    }),
    [failure]: (state, { payload: error = 'error' }) => ({
      ...state,
      requesting: false,
      error
    })
  },
  defaultState
)
