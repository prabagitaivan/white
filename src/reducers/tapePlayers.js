import { createSlice } from '@reduxjs/toolkit'

const initialState = {
  data: [],
  requesting: true,
  error: null
}

const slice = createSlice({
  name: 'tapePlayers',
  initialState,
  reducers: {
    request: state => ({ ...state, requesting: true, error: null }),
    edit: (state, { payload: data = [] }) => ({ ...state, data }),
    success: (state, { payload: data = [] }) => ({
      ...state,
      data,
      requesting: false,
      error: null
    }),
    failure: (state, { payload: error = 'error' }) => ({
      ...state,
      requesting: false,
      error
    })
  }
})

export const { request, edit, success, failure } = slice.actions

export default slice.reducer
