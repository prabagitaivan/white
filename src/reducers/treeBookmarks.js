import { createSlice } from '@reduxjs/toolkit'

const initialState = {
  data: {},
  requesting: true,
  error: null
}

const slice = createSlice({
  name: 'treeBookmarks',
  initialState,
  reducers: {
    request: state => ({ ...state, requesting: true, error: null }),
    success: (state, { payload: data = {} }) => ({
      ...state,
      data,
      requesting: false,
      error: null
    }),
    failure: (state, { payload: error = 'error' }) => ({
      ...state,
      requesting: false,
      error
    })
  }
})

export const { request, success, failure } = slice.actions

export default slice.reducer
