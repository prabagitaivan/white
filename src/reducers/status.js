import { createSlice } from '@reduxjs/toolkit'
import { getLightStatus } from '../libraries/styles'

const initialState = {
  online: true,
  desktop: true,
  light: getLightStatus(),
  page: 'RandomNotes'
}

const slice = createSlice({
  name: 'status',
  initialState,
  reducers: {
    setConnection: (state, { payload: online }) => ({ ...state, online }),
    setScreen: (state, { payload: size }) => ({
      ...state,
      desktop: size > 600
    }),
    setTheme: (state, { payload: light }) => ({ ...state, light }),
    setPage: (state, { payload: page }) => ({ ...state, page })
  }
})

export const { setConnection, setScreen, setTheme, setPage } = slice.actions

export default slice.reducer
