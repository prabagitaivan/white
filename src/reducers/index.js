import { combineReducers } from '@reduxjs/toolkit'
import status from './status'
import randomNotes from './randomNotes'
import treeBookmarks from './treeBookmarks'

const reducers = combineReducers({
  status,
  randomNotes,
  treeBookmarks
})

export default reducers
