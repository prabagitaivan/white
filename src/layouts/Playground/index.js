import React, { memo, useEffect, useState } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { Button, ButtonGroup, Typography, SvgIcon } from '@mui/material'
import { Clear, PlayArrow } from '@mui/icons-material'
import { makeStyles } from '@mui/styles'
import CodeMirror, { Prec, keymap } from '@uiw/react-codemirror'
import { javascript } from '@codemirror/lang-javascript'
import Navigator from '../../components/Navigator'
import Content from '../../components/Content'
import { setPage } from '../../reducers/status'
import logger from '../../libraries/logger'

const useStyles = makeStyles(
  {
    mainEditor: {
      marginTop: 20,
      fontSize: 14,
      border: '1px solid gray'
    },
    resultEditor: {
      fontSize: 14,
      border: '1px solid gray'
    },
    buttons: {
      marginTop: 15,
      marginBottom: 15,

      '& .MuiButton-root': {
        borderRadius: 0
      }
    },
    subButton: {
      flexDirection: 'row',
      justifyContent: 'center',
      width: '10vw'
    },
    subText: {
      fontSize: 14,
      marginTop: ({ desktop }) => (desktop ? -1.5 : 0),
      textTransform: 'none'
    },
    mainButton: {
      width: '80vw'
    },
    mainText: {
      fontSize: 14,
      marginLeft: 2,
      marginTop: ({ desktop }) => (desktop ? -1.5 : 0),
      textTransform: 'none'
    },
    keyText: {
      fontSize: 12,
      fontFamily: 'monospace',
      marginLeft: 10,
      marginTop: ({ desktop }) => (desktop ? -2 : 0),
      textTransform: 'none'
    }
  },
  { name: 'Playground' }
)

export default memo(() => {
  const { desktop, mac, light } = useSelector(state => state.status)
  const dispatch = useDispatch()
  const classes = useStyles({ desktop })
  const [code, setCode] = useState("console.log('hello world')")
  const [results, setResults] = useState([])

  const coding = value => setCode(value)

  const clear = () => {
    setCode('')
    setResults([])
  }

  const copy = () => {
    navigator.clipboard.writeText(code)
  }

  const onResults = sign => output => {
    const date = new Date().toISOString().slice(11)
    const data = `${date} ${sign} ${output}`

    setResults(state => [...state, data])
  }

  const execute = async () => {
    setResults([])

    try {
      // eslint-disable-next-line no-new-func
      new Function(code)()
    } catch (error) {
      onResults('ðŸ’¥')(error.toString())
    }
  }

  useEffect(() => {
    dispatch(setPage('Playground'))

    const originalFn = console.log
    console.log = logger(onResults('âœ¨'), originalFn)

    return () => {
      console.log = originalFn
    }
  }, [dispatch])

  return (
    <div>
      <Navigator />
      <Content>
        <CodeMirror
          autoFocus
          value={code}
          extensions={[
            javascript(),
            Prec.highest(
              keymap.of([
                { key: 'Mod-Enter', preventDefault: true, run: execute }
              ])
            )
          ]}
          onChange={coding}
          theme={light ? 'light' : 'dark'}
          height='calc(100vh - 25vh - 165px)'
          className={classes.mainEditor}
        />
        <ButtonGroup color='secondary' fullWidth className={classes.buttons}>
          <Button onClick={clear} className={classes.subButton}>
            <Clear />
            {desktop ? (
              <Typography className={classes.subText}>clear</Typography>
            ) : null}
          </Button>
          <Button onClick={copy} className={classes.subButton}>
            <SvgIcon>
              <path d='M 15 3.75 H 6 c -0.825 0 -1.5 0.675 -1.5 1.5 v 10.5 h 1.5 V 5.25 h 9 V 3.75 z m 2.25 3 H 9 c -0.825 0 -1.5 0.675 -1.5 1.5 v 10.5 c 0 0.825 0.675 1.5 1.5 1.5 h 8.25 c 0.825 0 1.5 -0.675 1.5 -1.5 V 8.25 c 0 -0.825 -0.675 -1.5 -1.5 -1.5 z m 0 12 H 9 V 8.25 h 8.25 v 10.5 z' />
            </SvgIcon>
            {desktop ? (
              <Typography className={classes.subText}>copy</Typography>
            ) : null}
          </Button>
          <Button onClick={execute} className={classes.mainButton}>
            <PlayArrow />
            <Typography className={classes.mainText}>execute</Typography>
            {desktop && (
              <Typography className={classes.keyText}>
                [{mac ? 'Cmd' : 'Ctrl'} + Enter]
              </Typography>
            )}
          </Button>
        </ButtonGroup>
        <CodeMirror
          editable={false}
          value={results.join('\n')}
          theme={light ? 'light' : 'dark'}
          height='25vh'
          className={classes.resultEditor}
        />
      </Content>
    </div>
  )
})
