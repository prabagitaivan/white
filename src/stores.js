import { createStore as ReduxCreateStore, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'
import reducers from './reducers'
import sagas from './sagas'

export const createStore = preloadedState => {
  const sagaMiddleware = createSagaMiddleware()
  const enhancer = applyMiddleware(sagaMiddleware)

  const store = ReduxCreateStore(reducers, preloadedState, enhancer)
  sagas.forEach(sagaMiddleware.run)

  return store
}

export default createStore()
