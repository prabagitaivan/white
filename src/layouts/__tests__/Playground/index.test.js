import { render, fireEvent, waitForDomChange } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from '../../../stores'
import Styles from '../../../styles'
import { setPage } from '../../../reducers/status'
import Playground from '../../Playground'

let store

function renderPlayground ({ desktop = true, light = true }) {
  const preloadedState = {
    status: { desktop, light, page: 'Playground' }
  }
  store = createStore(preloadedState)

  jest.spyOn(store, 'dispatch').mockReturnValue()
  jest
    .spyOn(Date.prototype, 'toISOString')
    .mockImplementation(() => '2001-01-01T01:01:01.001Z')

  return render(
    <Provider store={store}>
      <Styles>
        <Playground />
      </Styles>
    </Provider>
  )
}

afterEach(() => {
  store.dispatch.mockRestore()
  Date.prototype.toISOString.mockRestore()
})

describe('layouts Playground', () => {
  describe('snapshots', () => {
    it('contain navigator', () => {
      const { container } = renderPlayground({})
      const Navigator = container.querySelector('#navigator')
      expect(Navigator).toBeInTheDocument()
    })
    it('contain initial content in desktop', () => {
      const { container } = renderPlayground({})
      const Editors = container.getElementsByClassName('cm-activeLine cm-line')

      expect(Editors).toMatchSnapshot()
    })
    it('contain initial content in mobile', () => {
      const { container } = renderPlayground({ desktop: false })
      const Editors = container.getElementsByClassName('cm-activeLine cm-line')

      expect(Editors).toMatchSnapshot()
    })
    it('contain initial content in light mode', () => {
      const { container } = renderPlayground({})
      const Editors = container.getElementsByClassName('cm-activeLine cm-line')

      expect(Editors).toMatchSnapshot()
    })
    it('contain initial content in dark mode', () => {
      const { container } = renderPlayground({ light: false })
      const Editors = container.getElementsByClassName('cm-activeLine cm-line')

      expect(Editors).toMatchSnapshot()
    })
    it('contain buttons with text in desktop', () => {
      const { container } = renderPlayground({})
      const Content = container.querySelector('#content')

      expect(Content).toBeInTheDocument()
      expect(Content.childNodes[1]).toMatchSnapshot()
    })
    it('contain buttons without text in mobile', () => {
      const { container } = renderPlayground({ desktop: false })
      const Content = container.querySelector('#content')

      expect(Content).toBeInTheDocument()
      expect(Content.childNodes[1]).toMatchSnapshot()
    })
  })
  describe('mounting', () => {
    beforeEach(() => {
      renderPlayground({})
    })

    it('set page to playground after mount', () => {
      expect(store.dispatch).toHaveBeenCalledWith(setPage('Playground'))
    })
  })
  describe('userEvent', () => {
    beforeEach(() => {
      jest.spyOn(window.navigator.clipboard, 'writeText').mockReturnValue()
    })
    afterEach(() => {
      window.navigator.clipboard.writeText.mockRestore()
    })

    it('clear content when click clear button', async () => {
      const { container, getByText } = renderPlayground({})

      const Clear = getByText('clear')
      userEvent.click(Clear)

      const Editors = container.getElementsByClassName('cm-activeLine cm-line')
      await waitForDomChange(Editors)
      expect(Editors).toMatchSnapshot()
    })
    it('copy content when click copy button', () => {
      const { getByText } = renderPlayground({})

      const Copy = getByText('copy')
      userEvent.click(Copy)

      expect(window.navigator.clipboard.writeText).toHaveBeenCalledWith(
        "console.log('hello world')"
      )
    })
    it('execute the code content when click execute button', async () => {
      const { container, getByText } = renderPlayground({})

      const Execute = getByText('execute')
      userEvent.click(Execute)

      const Editors = container.getElementsByClassName('cm-activeLine cm-line')
      await waitForDomChange(Editors)
      expect(Editors).toMatchSnapshot()
    })
    it('execute the code with error content', async () => {
      const { container, getByText, findAllByRole } = renderPlayground({})

      const EditorTextBoxes = await findAllByRole('textbox')
      fireEvent.change(EditorTextBoxes[0], { target: { textContent: 'asdf' } })

      const Editors = container.getElementsByClassName('cm-activeLine cm-line')
      await waitForDomChange(Editors)

      const Execute = getByText('execute')
      userEvent.click(Execute)

      await waitForDomChange(Editors)
      expect(Editors).toMatchSnapshot()
    })
  })
})
