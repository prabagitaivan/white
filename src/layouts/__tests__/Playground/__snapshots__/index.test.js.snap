// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts Playground snapshots contain buttons with text in desktop 1`] = `
<div
  class="MuiButtonGroup-root WhitePlayground-buttons MuiButtonGroup-fullWidth"
  role="group"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined WhitePlayground-subButton MuiButton-fullWidth"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <p
        class="MuiTypography-root WhitePlayground-subText WhitePlayground-subText MuiTypography-body1"
      >
        clear
      </p>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined WhitePlayground-subButton MuiButton-fullWidth"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M 15 3.75 H 6 c -0.825 0 -1.5 0.675 -1.5 1.5 v 10.5 h 1.5 V 5.25 h 9 V 3.75 z m 2.25 3 H 9 c -0.825 0 -1.5 0.675 -1.5 1.5 v 10.5 c 0 0.825 0.675 1.5 1.5 1.5 h 8.25 c 0.825 0 1.5 -0.675 1.5 -1.5 V 8.25 c 0 -0.825 -0.675 -1.5 -1.5 -1.5 z m 0 12 H 9 V 8.25 h 8.25 v 10.5 z"
        />
      </svg>
      <p
        class="MuiTypography-root WhitePlayground-subText WhitePlayground-subText MuiTypography-body1"
      >
        copy
      </p>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined WhitePlayground-mainButton MuiButton-fullWidth"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8 5v14l11-7z"
        />
      </svg>
      <p
        class="MuiTypography-root WhitePlayground-mainText WhitePlayground-mainText MuiTypography-body1"
      >
        execute
      </p>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`layouts Playground snapshots contain buttons without text in mobile 1`] = `
<div
  class="MuiButtonGroup-root WhitePlayground-buttons MuiButtonGroup-fullWidth"
  role="group"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined WhitePlayground-subButton MuiButton-fullWidth"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined WhitePlayground-subButton MuiButton-fullWidth"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M 15 3.75 H 6 c -0.825 0 -1.5 0.675 -1.5 1.5 v 10.5 h 1.5 V 5.25 h 9 V 3.75 z m 2.25 3 H 9 c -0.825 0 -1.5 0.675 -1.5 1.5 v 10.5 c 0 0.825 0.675 1.5 1.5 1.5 h 8.25 c 0.825 0 1.5 -0.675 1.5 -1.5 V 8.25 c 0 -0.825 -0.675 -1.5 -1.5 -1.5 z m 0 12 H 9 V 8.25 h 8.25 v 10.5 z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined WhitePlayground-mainButton MuiButton-fullWidth"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8 5v14l11-7z"
        />
      </svg>
      <p
        class="MuiTypography-root WhitePlayground-mainText WhitePlayground-mainText MuiTypography-body1"
      >
        execute
      </p>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`layouts Playground snapshots contain initial content in dark mode 1`] = `
HTMLCollection [
  <div
    class="cm-activeLine cm-line"
  >
    <span
      class="Í¼q"
    >
      console
    </span>
    <span
      class="Í¼v"
    >
      .
    </span>
    <span
      class="Í¼q"
    >
      log
    </span>
    (
    <span
      class="Í¼13"
    >
      'hello world'
    </span>
    )
  </div>,
  <div
    class="cm-activeLine cm-line"
  >
    <br />
  </div>,
]
`;

exports[`layouts Playground snapshots contain initial content in desktop 1`] = `
HTMLCollection [
  <div
    class="cm-activeLine cm-line"
  >
    console.log(
    <span
      class="Í¼e"
    >
      'hello world'
    </span>
    )
  </div>,
  <div
    class="cm-activeLine cm-line"
  >
    <br />
  </div>,
]
`;

exports[`layouts Playground snapshots contain initial content in light mode 1`] = `
HTMLCollection [
  <div
    class="cm-activeLine cm-line"
  >
    console.log(
    <span
      class="Í¼e"
    >
      'hello world'
    </span>
    )
  </div>,
  <div
    class="cm-activeLine cm-line"
  >
    <br />
  </div>,
]
`;

exports[`layouts Playground snapshots contain initial content in mobile 1`] = `
HTMLCollection [
  <div
    class="cm-activeLine cm-line"
  >
    console.log(
    <span
      class="Í¼e"
    >
      'hello world'
    </span>
    )
  </div>,
  <div
    class="cm-activeLine cm-line"
  >
    <br />
  </div>,
]
`;

exports[`layouts Playground userEvent clear content when click clear button 1`] = `
HTMLCollection [
  <div
    class="cm-activeLine cm-line"
  >
    <br />
  </div>,
  <div
    class="cm-activeLine cm-line"
  >
    <br />
  </div>,
]
`;

exports[`layouts Playground userEvent execute the code content when click execute button 1`] = `
HTMLCollection [
  <div
    class="cm-activeLine cm-line"
  >
    console.log(
    <span
      class="Í¼e"
    >
      'hello world'
    </span>
    )
  </div>,
  <div
    class="cm-activeLine cm-line"
  >
    01:01:01.001Z âœ¨ "hello world"
  </div>,
]
`;

exports[`layouts Playground userEvent execute the code with error content 1`] = `
HTMLCollection [
  <div
    class="cm-activeLine cm-line"
  >
    asdf
  </div>,
  <div
    class="cm-activeLine cm-line"
  >
    01:01:01.001Z ðŸ’¥ ReferenceError: asdf is not defined
  </div>,
]
`;
